x-pack/plugins/actions/server/authorization/should_legacy_rbac_apply_by_source.test.ts

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { shouldLegacyRbacApplyBySource } from './should_legacy_rbac_apply_by_source';
+import { savedObjectsClientMock } from '../../../../../src/core/server/mocks';
+import uuid from 'uuid';
+import { asSavedObjectExecutionSource } from '../lib';
+
+const unsecuredSavedObjectsClient = savedObjectsClientMock.create();
+
+describe(`#shouldLegacyRbacApplyBySource`, () => {
+  test('should return false if no source is provided', async () => {
+    expect(await shouldLegacyRbacApplyBySource(unsecuredSavedObjectsClient)).toEqual(false);
+  });
+
+  test('should return false if source is not an alert', async () => {
+    expect(
+      await shouldLegacyRbacApplyBySource(
+        unsecuredSavedObjectsClient,
+        asSavedObjectExecutionSource({
+          type: 'action',
+          id: uuid.v4(),
+        })
+      )
+    ).toEqual(false);
+  });
+
+  test('should return false if source alert is not marked as legacy', async () => {
+    const id = uuid.v4();
+    unsecuredSavedObjectsClient.get.mockResolvedValue(mockAlert({ id }));
+    expect(
+      await shouldLegacyRbacApplyBySource(
+        unsecuredSavedObjectsClient,
+        asSavedObjectExecutionSource({
+          type: 'alert',
+          id,
+        })
+      )
+    ).toEqual(false);
+  });
+
+  test('should return true if source alert is marked as legacy', async () => {
+    const id = uuid.v4();
+    unsecuredSavedObjectsClient.get.mockResolvedValue(
+      mockAlert({ id, attributes: { meta: { versionApiKeyLastmodified: 'pre-7.10.0' } } })
+    );
+    expect(
+      await shouldLegacyRbacApplyBySource(
+        unsecuredSavedObjectsClient,
+        asSavedObjectExecutionSource({
+          type: 'alert',
+          id,
+        })
+      )
+    ).toEqual(true);
+  });
+
+  test('should return false if source alert is marked as modern', async () => {
+    const id = uuid.v4();
+    unsecuredSavedObjectsClient.get.mockResolvedValue(
+      mockAlert({ id, attributes: { meta: { versionApiKeyLastmodified: '7.10.0' } } })
+    );
+    expect(
+      await shouldLegacyRbacApplyBySource(
+        unsecuredSavedObjectsClient,
+        asSavedObjectExecutionSource({
+          type: 'alert',
+          id,
+        })
+      )
+    ).toEqual(false);
+  });
+
+  test('should return false if source alert is marked with a last modified version', async () => {
```

Comment:
It doesn't seem like having `meta: {}` is something that would ever occur in practice, but it's fine as a test none-the-less since ... who knows, I guess it could happen (or maybe I'm wrong and it does occur in practice).  The name of the test seems odd though - does having a `meta` with no `versionApiKeyLastmodified` imply "marked with a last modified version"?  I might have missed how this factors in, would have guessed that this would be treated the same as if `meta` wasn't supplied at all.

Reply:
Yeah, name got mangled up.
Basically, we're testing to see that if we don't have a `versionApiKeyLastmodified` then we use full RBAC.

Renamed 👍